tests/cli-rustup.rs

```diff
@@ -803,6 +803,33 @@ fn list_default_and_override_toolchain() {
     });
 }
 
+#[test]
+fn heal_damaged_toolchain() {
+    setup(&|config| {
+        expect_ok(config, &["rustup", "default", "nightly"]);
+        expect_ok(config, &["rustup", "show", "active-toolchain"]);
```

Comment:
When I suggested adding this check, it was because we need to ensure that when the toolchain doesn't need healing we don't try syncing the channel once more.  As such, something like the following would make sense:

```suggestion
        expect_not_stderr_ok(config, &["rustup", "show", "active-toolchain"], "syncing channel updates");
```

Reply:
@kinnison I appreciate the patient support and yes, I would like to finish this.